src/Kunstmaan/NodeSearchBundle/DependencyInjection/Configuration.php

```diff
@@ -28,16 +29,35 @@ public function getConfigTreeBuilder()
         /** @var ArrayNodeDefinition $properties */
         $properties = $rootNode->children()->arrayNode('mapping')->useAttributeAsKey('name')->prototype('array');
 
-        $properties->children()->scalarNode('type')->beforeNormalization()->ifNotInArray($types = [
-            'string', 'token_count', 'text', 'keyword',
+        $types = [
+            'token_count', 'text', 'keyword',
             'float', 'double', 'byte', 'short', 'integer', 'long',
             'date',
             'boolean',
             'binary',
-        ])->thenInvalid('type must be one of: ' . implode(', ', $types));
-        $properties->children()->scalarNode('index')->beforeNormalization()->ifNotInArray(['analyzed', 'not_analyzed', 'no'])
-            ->thenInvalid("index must be one of: analyzed, not_analyzed, no");
-        $properties->children()->booleanNode('include_in_all');
+        ];
+        if (!ElasticSearchUtil::useVersion6()) {
+            $types[] = 'string';
+        }
+
+        $properties->children()->scalarNode('type')->beforeNormalization()->ifNotInArray($types)->thenInvalid('type must be one of: ' . implode(', ', $types));
+
+        if (ElasticSearchUtil::useVersion6()) {
+            $properties->children()->booleanNode('fielddata');
```

Comment:
I was reading the https://www.elastic.co/guide/en/elasticsearch/reference/current/fielddata.html article and like they say, it's not a good idea to use this fielddata element because it will cause a lot of memory issues. We should use https://www.elastic.co/guide/en/elasticsearch/reference/current/doc-values.html.

Reply:
yes, using doc_values seems a good alternative, however, I don't think we should prohibit it. We are not using fielddata, but allowing that it is used as a mapping value. I'll add the doc_values as well.

Reply:
Alright, super